File: JavaScript_Basic_04_2023_SolutionsOfProblems/03.MoreEx_PB-JS-Nested-Conditional-Statements/P01_MatchTickets.js
function matchTickets(input) {
    let budget = Number(input[0]);
    let ticketType = input[1];
    let peopleNumber = Number(input[2]);

    let ticketBudget = 0.0;

    if (peopleNumber >= 50) {
        ticketBudget = budget * 0.75;
    } else if (peopleNumber >= 25) {
        ticketBudget = budget * 0.6;
    } else if (peopleNumber >= 10) {
        ticketBudget = budget * 0.5;
    } else if (peopleNumber >= 5) {
        ticketBudget = budget * 0.4;
    } else {
        ticketBudget = budget * 0.25;
    }

    if (ticketType === "VIP") {
        if (peopleNumber * 499.99 > ticketBudget) {
            console.log(`Not enough money! You need ${(peopleNumber * 499.99 - ticketBudget).toFixed(2)} leva.`);
        } else {
            console.log(`Yes! You have ${(ticketBudget - peopleNumber * 499.99).toFixed(2)} leva left.`);
        }
    } else {
        if (peopleNumber * 249.99 > ticketBudget) {
            console.log(`Not enough money! You need ${(peopleNumber * 249.99 - ticketBudget).toFixed(2)} leva.`);
        } else {
            console.log(`Yes! You have ${(ticketBudget - peopleNumber * 249.99).toFixed(2)} leva left.`);
        }
    }
}

matchTickets([30000,
    "VIP",
    49
])

File: Java_Basic_Jan_2023_SolutionsOfProblems/P03AdditionalProblems_ConditionalStatementsAdvanced/P01_MatchTickets.java
package P03AdditionalProblems_ConditionalStatementsAdvanced;

import java.util.Scanner;

public class P01_MatchTickets {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        double budget = Double.parseDouble(scanner.nextLine());
        String ticketType = scanner.nextLine();
        int numberPeople = Integer.parseInt(scanner.nextLine());

        double transportPrice = 0;
        double ticketPrice = 0;
        if (ticketType.equals("VIP")) {
            ticketPrice = numberPeople * 499.99;
        } else {
            ticketPrice = numberPeople * 249.99;
        }
        if (numberPeople >= 1 && numberPeople <= 4) {
            transportPrice = budget * 0.75;
        } else if (numberPeople <= 9) {
            transportPrice = budget * 0.6;
        } else if (numberPeople <= 24) {
            transportPrice = budget * 0.5;
        } else if (numberPeople <= 49) {
            transportPrice = budget * 0.4;
        } else {
            transportPrice = budget * 0.25;
        }
        if ((transportPrice + ticketPrice) > budget) {
            System.out.printf("Not enough money! You need %.2f leva.", transportPrice + ticketPrice - budget);
        } else {
            System.out.printf("Yes! You have %.2f leva left.", budget - transportPrice - ticketPrice);
        }

    }
}
